src/constraint_system/ecc/scalar_mul/variable_base/mod.rs

```diff
@@ -0,0 +1,164 @@
+use crate::constraint_system::ecc::{Point, PointScalar};
+use crate::constraint_system::{variable::Variable, StandardComposer};
+use dusk_bls12_381::Scalar;
+
+/// Computes a Scalar multiplication with the input scalar and a chosen generator
+pub fn variable_base_scalar_mul(
+    composer: &mut StandardComposer,
+    jubjub_var: Variable,
+    point: Point,
+) -> PointScalar {
+    // Turn scalar into bits
+    let raw_bls_scalar = *composer.variables.get(&jubjub_var).unwrap();
+    let scalar_bits_var = scalar_decomposition(composer, jubjub_var, raw_bls_scalar);
+
+    let identity = Point::identity(composer);
+    let mut result = identity;
+
+    for bit in scalar_bits_var.into_iter().rev() {
+        result = result.fast_add(composer, result);
+        let point_to_add = conditional_select_identity(composer, bit, point);
+        result = result.fast_add(composer, point_to_add);
+    }
+
+    PointScalar {
+        point: result,
+        scalar: jubjub_var,
+    }
+}
+
+/// If bit == 0, then return zero else return value
+/// This is the polynomial f(x) = x * a
+/// Where x is the bit
+fn conditional_select_zero(
+    composer: &mut StandardComposer,
+    bit: Variable,
+    value: Variable,
+) -> Variable {
+    // returns bit * value
+    composer.mul(Scalar::one(), bit, value, Scalar::zero(), Scalar::zero())
+}
+/// If bit == 0, then return 1 else return value
+/// This is the polynomial f(x) = 1 - x + xa
+/// Where x is the bit
+fn conditional_select_one(
+    composer: &mut StandardComposer,
+    bit: Variable,
+    value: Variable,
+) -> Variable {
+    let value_scalar = composer.variables.get(&value).unwrap();
+    let bit_scalar = composer.variables.get(&bit).unwrap();
+
+    let f_x_scalar = Scalar::one() - bit_scalar + (bit_scalar * value_scalar);
+    let f_x = composer.add_input(f_x_scalar);
+
+    composer.poly_gate(
+        bit,
+        value,
+        f_x,
+        Scalar::one(),
+        -Scalar::one(),
+        Scalar::zero(),
+        -Scalar::one(),
+        Scalar::one(),
+        Scalar::zero(),
+    );
+
+    f_x
+}
```

Comment:
This is already in `plonk_gadgets` too. But it takes more constraints there.

My suggestion would be to port this to `plonk_gadgets` and then get the function from there.
WDYT @ZER0 @vlopes11 @kevaundray ?

Reply:
Same as above, if this method is needed in the future, it can be made public. Feel free to open a new issue, regarding this